client/GroupEmote.lua

```diff
@@ -52,15 +55,19 @@ RegisterNetEvent("rpemotes:client:startGroupEmote", function(reqid, zone)
 end)
 
 AddEventHandler("rpemotes:client:autoCancel", function()
+
+    -- We don't want to auto-cancel the group emote request, if the player already has an emote playing.
+    local existingEmote = LocalPlayer.state.currentEmote 
+
     while groupEmoteReqId ~= nil and groupEmoteAccepted == true do
-        -- Keep rechecking to make sure the player is still in area.
+        -- Keep rechecking to make sure the player is still in area and not using another emote.
+        local currentEmote = LocalPlayer.state.currentEmote
         local crds = GetEntityCoords(PlayerPedId())
-        if #(crds - (groupEmoteOriginCoords or vector3(0))) > groupEmoteOriginRadius or LocalPlayer.state.currentEmote ~= nil then
+        if #(crds - (groupEmoteOriginCoords or vector3(0))) > groupEmoteOriginRadius or (currentEmote ~= nil and currentEmote ~= existingEmote) then
```

Comment:
On second thought, I think we should allow players to emote in the meantime but then be forced into the group emote. I'm imagining a scenario where players are dancing. It would be a bit weird if we forced them to sit there for 10 seconds and do nothing in the meantime. I think it's better if they can still do other emotes, but just can't leave the area.